servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractRestNamespace.java

```diff
@@ -321,28 +338,37 @@ public void testNamespacesWithAndWithoutZeroBytes() throws BaseNessieClientServe
     String firstName = "a.b\u0000c.d";
     String secondName = "a.b.c.d";
 
-    Namespace first = Namespace.parse(firstName);
-    Namespace second = Namespace.parse(secondName);
-    List<Namespace> namespaces = Arrays.asList(first, second);
-
     // perform creation and retrieval
-    for (Namespace namespace : namespaces) {
-      Namespace created =
-          getApi().createNamespace().refName(branch.getName()).namespace(namespace).create();
-      assertThat(created).isNotNull().isEqualTo(namespace);
-      assertThat(getApi().getNamespace().refName(branch.getName()).namespace(namespace).get())
-          .isEqualTo(created);
-
-      assertThatThrownBy(
-              () ->
-                  getApi()
-                      .createNamespace()
-                      .refName(branch.getName())
-                      .namespace(namespace)
-                      .create())
-          .isInstanceOf(NessieNamespaceAlreadyExistsException.class)
-          .hasMessage(String.format("Namespace '%s' already exists", namespace.name()));
-    }
+    ThrowingExtractor<String, Namespace, ?> creator =
+        identifier -> {
+          Namespace namespace = Namespace.parse(identifier);
+
+          Namespace created =
+              getApi().createNamespace().refName(branch.getName()).namespace(namespace).create();
+          assertThat(created)
+              .isNotNull()
+              .extracting(Namespace::getElements, Namespace::toPathString)
+              .containsExactly(namespace.getElements(), namespace.toPathString());
+
+          assertThat(getApi().getNamespace().refName(branch.getName()).namespace(namespace).get())
+              .isEqualTo(created);
+
+          assertThatThrownBy(
+                  () ->
+                      getApi()
+                          .createNamespace()
+                          .refName(branch.getName())
+                          .namespace(namespace)
+                          .create())
+              .isInstanceOf(NessieNamespaceAlreadyExistsException.class)
+              .hasMessage(String.format("Namespace '%s' already exists", namespace.name()));
+
+          return created;
+        };
+
+    Namespace first = creator.apply(firstName);
```

Comment:
also not sure why these changes are required. The only necessary change it seems is the comparison of `Namespace::getElements, Namespace::toPathString`

Reply:
Also to not mess with all the assertions below - to have the content-ID